csrc/scheduler/resize.cpp

```diff
@@ -133,6 +120,44 @@ bool ResizeScheduler::canScheduleCompileTime(Fusion* fusion) {
     return false;
   }
 
+  // Having different resizes between outputs is not allowed at this
+  // moment. For example, consider a fusion like:
+  //
+  // t0 = [i0]
+  // fusion.addInput(t0)
+  // t1 = t0[:i0/2]
+  // t2 = t0[i0/2:]
+  // fusion.addOutput(t1)
+  // fusion.addOutput(t2)
+  //
+  // For now, this is not going to be fused since t1 and t2 have
+  // different resize ops, although in this case, since the extents of t1 and
+  // t2 are the same, it should be relatively straightforward to fuse them
+  // together.
+  for (auto out_tv : ir_utils::filterByType<TensorView>(fusion->outputs())) {
+    if (out_tv == ref_tv) {
+      continue;
+    }
+    auto exprs = ValGraphBFS::getExprGroupsBetween(
+                     broadcast_graph,
+                     broadcast_graph.toGroups(ref_tv->getLogicalDomain()),
+                     broadcast_graph.toGroups(out_tv->getLogicalDomain()),
+                     /*require_all_to_visited=*/false)
+                     .first;
+    for (const auto& [expr_g, dir] : exprs) {
+      if (expr_g->front()->isA<Resize>()) {
```

Comment:
Is my understanding correct that we will segment if we have one resized output and one not resized?
```
addInput(tv0)
tv1 = 2 * tv0;
tv2 = slice(tv1);
tv3 = 3 * tv2;
addOutput(tv2);
addOutput(tv3);
```
In that case we will still have a resize between the `tv2` and `tv3` but it seems like we would potentially be able to schedule it like `tv3`.

Reply:
Yes, it is definitely possible, but it's a non-trivial problem to pick a reference tensor. If the dimension of `tv2` is not that different from `tv1`, either of `tv2` or `tv3` should be fine. However, if if they are significantly different, it's unclear if we should fuse them or not. For a trivial fusion like this, it's definitely better to fuse. For more complex fusions, we may be able to generate more efficient segmented kernels. At this point, this is not something I'm trying to address.